### PR TITLE
Feat/System deprecation

### PR DESCRIPTION
**Pull request by ElliotFriedman**

### CODE REVIEW

File: contracts/test/integration/vip/vip15.sol [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+
+    /// @notice target token balance for the USDC PSM to hold
+    uint248 private constant targetBalanceUsdc = 100_000e6;
+
+    /// @notice scale up USDC value by 12 decimals in order to account for decimal delta
+    /// and properly update the buffer in ERC20Allocator
+    int8 private constant usdcDecimalNormalizer = 12;
+
+    /// @notice no scaling to do on DAI as decimals are 18
+    int8 private constant daiDecimalNormalizer = 0;
```
Comment (ElliotFriedman) [inline contracts/test/integration/vip/vip15.sol]: delete

File: contracts/test/integration/vip/vip15.sol [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+                description: "Pause USDC PSM"
+            })
+        );
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.VOLT_DAI_PSM,
+                arguments: abi.encodeWithSignature("pause()"),
+                description: "Pause DAI Morpho Compound PCV Deposit"
```
Comment (eswak) [inline contracts/test/integration/vip/vip15.sol]:
```suggestion
                description: "Pause DAI PSM"
```

File: contracts/test/integration/vip/vip15.sol [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.CORE,
+                arguments: abi.encodeWithSignature(
+                    "revokePCVController(address)",
+                    MainnetAddresses.ERC20ALLOCATOR
+                ),
+                description: "Revoke PCV Controller from ERC20 Allocator"
+            })
+        );
```
Comment (eswak) [inline contracts/test/integration/vip/vip15.sol:242]:
```suggestion
        );

        proposal.push(
            ITimelockSimulation.action({
                value: 0,
                target: MainnetAddresses.CORE,
                arguments: abi.encodeWithSignature(
                    "revokePCVController(address)",
                    MainnetAddresses.PCV_GUARDIAN
                ),
                description: "Revoke PCV Controller from PCV_GUARDIAN"
            })
        );

        proposal.push(
            ITimelockSimulation.action({
                value: 0,
                target: MainnetAddresses.CORE,
                arguments: abi.encodeWithSignature(
                    "revokePCVController(address)",
                    MainnetAddresses.GOVERNOR
                ),
                description: "Revoke PCV Controller from Multisig"
            })
        );
```

Also revoke PCV_CONTOLLER from PCVGuardian & Multisig

File: contracts/test/integration/vip/vip15.sol [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.CORE,
+                arguments: abi.encodeWithSignature(
+                    "revokeGovernor(address)",
+                    MainnetAddresses.GOVERNOR
+                ),
+                description: "Revoke PCV Controller from ERC20 Allocator"
```
Comment (eswak) [inline contracts/test/integration/vip/vip15.sol:252]:
```suggestion
                    MainnetAddresses.TIMELOCK_CONTROLLER
                ),
                description: "Revoke GOVERNOR from Timelock"
```

Keep `GOVERNOR` role on Multisig, if needed for whatever cleanup item later. The system won't hold any user funds anyway. Clean it from the Timelock.

Additionally, to completely clean up roles, you should also revoke :
- `GUARDIAN` role from `PCV_GUARDIAN`
- `PCV_GUARD_ADMIN_ROLE` from `PCV_GUARD_ADMIN`
- `PCV_GUARD_ROLE` role from `EOA_1`, `EOA_2`, and `EOA_4`

In the upgraded system, these roles have to be granted with the new CoreV2

**Review by eswak — CHANGES_REQUESTED** [review]
Some changes to make on the role deprecation

What would you think of prefixing all contracts that are deprecated in `MainnetAddresses.sol` with `V1__*` ? This will allow us, upon deploying the new system, to populate `MainnetAddresses.sol` with clean constant names. We can also do this change in the PR that deploys the new system, but it feels like it should be part of deprecation

File: contracts/test/integration/vip/vip15.sol [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+        override
+        returns (ITimelockSimulation.action[] memory prop)
+    {
+        prop = proposal;
+    }
+
+    function mainnetSetup() public override {}
+
+    function mainnetValidate() public override {
+        assertTrue(!core.isPCVController(address(allocator)));
```
Comment (ElliotFriedman) [inline contracts/test/integration/vip/vip15.sol:511]: add additional assertions here that all state transitions that we wanted to happen happened.

**Comment by ElliotFriedman** [comment]
> What would you think of prefixing all contracts that are deprecated in `MainnetAddresses.sol` with `V1__*` ? This will allow us, upon deploying the new system, to populate `MainnetAddresses.sol` with clean constant names. We can also do this change in the PR that deploys the new system, but it feels like it should be part of deprecation

Let's do that in a separate PR. This doesn't need to be a blocker for the audit.

File: contracts/test/integration/vip/vip15.sol [hunk 6]
```diff
@@ -85,6 +68,20 @@ contract vip15 is DSTest, IVIP {
             return;
         }
 
+        /// ------- poker morpho to update p2p indexes -------
```
Comment (eswak) [inline contracts/test/integration/vip/vip15.sol]: ♠️ ♦️ ♣️ ♥️

File: contracts/test/integration/vip/vip15.sol [hunk 7]
```diff
@@ -256,6 +253,91 @@ contract vip15 is DSTest, IVIP {
...
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.PCV_GUARD_ADMIN,
+                arguments: abi.encodeWithSignature(
+                    "revokePCVGuardRole(address)",
+                    MainnetAddresses.EOA_1
+                ),
+                description: "Revoke Guardian from EOA_1"
```
Comment (eswak) [inline contracts/test/integration/vip/vip15.sol]:
```suggestion
                description: "Revoke PCV_GUARD from EOA_1"
```

File: contracts/test/integration/vip/vip15.sol [hunk 8]
```diff
@@ -256,6 +253,91 @@ contract vip15 is DSTest, IVIP {
...
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.PCV_GUARD_ADMIN,
+                arguments: abi.encodeWithSignature(
+                    "revokePCVGuardRole(address)",
+                    MainnetAddresses.EOA_2
+                ),
+                description: "Revoke Guardian from EOA_2"
```
Comment (eswak) [inline contracts/test/integration/vip/vip15.sol]:
```suggestion
                description: "Revoke PCV_GUARD from EOA_2"
```

File: contracts/test/integration/vip/vip15.sol [hunk 9]
```diff
@@ -256,6 +253,91 @@ contract vip15 is DSTest, IVIP {
...
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.PCV_GUARD_ADMIN,
+                arguments: abi.encodeWithSignature(
+                    "revokePCVGuardRole(address)",
+                    MainnetAddresses.EOA_4
+                ),
+                description: "Revoke Guardian from EOA_4"
```
Comment (eswak) [inline contracts/test/integration/vip/vip15.sol]:
```suggestion
                description: "Revoke PCV_GUARD from EOA_4"
```

File: contracts/test/integration/vip/vip15.sol [hunk 10]
```diff
@@ -416,17 +498,45 @@ contract vip15 is DSTest, IVIP {
...
 
     function mainnetValidate() public override {
+        /// core roles
+        assertEq(core.getRoleMemberCount(core.PCV_CONTROLLER_ROLE()), 0);
+        assertEq(core.getRoleMemberCount(core.GUARDIAN_ROLE()), 0);
+        assertEq(core.getRoleMemberCount(GOVERN_ROLE), 2);
+        assertEq(core.getRoleMemberCount(core.MINTER_ROLE()), 0);
+        assertEq(core.getRoleMemberCount(core.BURNER_ROLE()), 0);
+        assertEq(core.getRoleMemberCount(PCV_GUARD_ADMIN_ROLE), 0);
+        assertEq(core.getRoleMemberCount(VoltRoles.PCV_GUARD), 0);
```
Comment (eswak) [inline contracts/test/integration/vip/vip15.sol:508]: perfect

File: contracts/test/integration/vip/vip15.sol (hunk 3 above)
Comment (ElliotFriedman) [reply contracts/test/integration/vip/vip15.sol:242]: this change has been integrated into the PR

File: contracts/test/integration/vip/vip15.sol (hunk 4 above)
Comment (ElliotFriedman) [reply contracts/test/integration/vip/vip15.sol:252]: all roles have been revoked, however, timelock has governor so we can always make an update to the system if needed through proposing via the multisig.

**Review by eswak — APPROVED** [review]

**Pull request merged**